File: algorithms/1_easy/D/01108_defanging_an_ip_address/solution.cpp
class Solution {
public:
    string defangIPaddr(string address) {
        string res = "";

        for (const char &ch: address) {
            if (ch == '.') {
                res += "[.]";
            } else {
                res.push_back(ch);
            }
        }

        return res;
    }
};

File: algorithms/1_easy/D/01108_defanging_an_ip_address/solution.rs
impl Solution {
    pub fn defang_i_paddr(address: String) -> String {
        let mut res: Vec<u8> = Vec::with_capacity(address.len());

        for byte in address.into_bytes().into_iter() {
            if byte == b'.' {
                res.extend([b'[', b'.', b']']);
            } else {
                res.push(byte);
            }
        }
        
        String::from_utf8(res).unwrap()
    }
}
